samples/Kconfig

```diff
@@ -16,15 +16,134 @@ config NCS_SAMPLES_DEFAULTS
 	help
 	  Add the empty_app_core as a child image for the given sample.
 	  Used by samples that only require SOC_NRF5340_CPUNET.
 
+config NCS_SAMPLE_MULTIPROTOCOL_RPMSG_CHILD_IMAGE
+	bool "Add multiprotocol_rpmsg sample to network core"
+	default y if BT_RPMSG_NRF53 && NRF_802154_SER_HOST
+	imply AUTO_INCLUDE_CHILD_IMAGE
+	help
+	  Add the multiprotocal_rpmsg sample as a child image on the network
+	  core for the given sample.
+
+config NCS_SAMPLE_HCI_RPMSG_CHILD_IMAGE
+	bool "Add hci_rpmsg sample to network core"
+	default y if BT_RPMSG_NRF53
+	imply AUTO_INCLUDE_CHILD_IMAGE
+	help
+	  Add the hci_rpmsg sample as a child image on the network core for the
+	  given sample.
+
+config NCS_SAMPLE_802154_RPMSG_CHILD_IMAGE
+	bool "Add 802154_rpmsg sample to network core"
+	default y if NRF_802154_SER_HOST
+	imply AUTO_INCLUDE_CHILD_IMAGE
+	help
+	  Add the 802154_rpmsg sample as a child image on the network core for
+	  the given sample.
+
+config NCS_SAMPLE_RPC_HOST_CHILD_IMAGE
+	bool "Add rpc_host sample to network core"
+	default y if BT_RPC
+	imply AUTO_INCLUDE_CHILD_IMAGE
+	help
+	  Add the rpc_host sample as a child image on the network core for the
+	  given sample.
+
+config AUTO_INCLUDE_CHILD_IMAGE
+	bool
+	select NCS_ADD_CHILD_IMAGE_TO_OTHER_CORE
+	help
+	  Helper configuration to indicate that an automatic inclusion of
+	  a child image is performed. Some prompts are removed if this
+	  option is set to ensure that the user does not interfere with the
+	  automatic inclusion.
+
+config NCS_ADD_CHILD_IMAGE_TO_OTHER_CORE
+	bool
+	prompt "Add child image to the application/network core" if !AUTO_INCLUDE_CHILD_IMAGE
+	imply PARTITION_MANAGER_ENABLED
+	help
+	  Add a child image to the other core. If the current sample is built
+	  for the APP core the child image will be build for the NET core and
+	  vice versa. To allow skipping build of the specified child image
+	  include the build_strategy_domain Kconfig template (this is handled
+	  automatically if AUTO_INCLUDE_CHILD_IMAGE is set).
+
+if NCS_ADD_CHILD_IMAGE_TO_OTHER_CORE
+
+config NCS_DOMAIN_TO_ADD_CHILD_IMAGE_TO
+	string
+	default "CPUNET" if SOC_NRF5340_CPUAPP
+	default "CPUAPP" if SOC_NRF5340_CPUNET
+	help
+	  The name of the domain to which the child image will be added.
+
+config NCS_CHILD_IMAGE_NAME
+	string
+	prompt  "Child image name" if !AUTO_INCLUDE_CHILD_IMAGE
+	default "empty_app_core" if NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE
+	default "multiprotocol_rpmsg" if NCS_SAMPLE_MULTIPROTOCOL_RPMSG_CHILD_IMAGE
+	default "hci_rpmsg" if NCS_SAMPLE_HCI_RPMSG_CHILD_IMAGE
+	default "802154_rpmsg" if NCS_SAMPLE_802154_RPMSG_CHILD_IMAGE
+	default "rpc_host" if NCS_SAMPLE_RPC_HOST_CHILD_IMAGE
+	default "hello_world"  # To stop it from failing in default case
+	help
+	  The name of the child image. This must match the name passed to the
+	  build strategy template.
+
+config NCS_CHILD_IMAGE_PATH
+	string
+	prompt  "Path to child image" if !AUTO_INCLUDE_CHILD_IMAGE
```

Comment:
as discussed.

Consider remove the path content, and instead make it just a symbol, and then CMake knows how to relate that to a path.

Should also make it easier to then create a choice menu for easy selection.

A user opting-out, i.e. `AUTO_INCLUDE_CHILD_IMAGE=n` must then use `add_child_image()` in CMake in his own project / Zephyr module.

We should supply an example on how a sample can easily be made into a local Zephyr module, as `add_child_image()` with partition manager only works correctly when called as part of boilerplate code.
(outside the scope of this PR)

Reply:
as haakon doesn't have the time to continue with this pull request, and since I asked him for this functionality anyways, I will continue with it.
I removed the path prompt and updated the commit message, does the new way seem ok to you ?

Tested with central_hr sample, the options don't appear in Kconfig any more and I am able to use add_child_image() in my app's CMakeLists.txt to compile a custom one for the network core.

Reply:
@jori-nordic Great to hear. Will take a look.